Restore widget theme and intents on update

Trogel opened the pull request:
For me this change resolves issue #31. It essentially makes sure to set the widget theme and create its intents also whenever the More/LessWidgetService runs onHandleIntent(). This might not be the best possible solution, but it appears to have the desired effect. And so far it does not show bad side-effects.

Hanzei commented:
Can we merge this?

BrainBlasted commented:
Found one issue using this version: New widgets were completely blank

BrainBlasted commented:
I spoke too soon. It seems I just needed to restart my launcher for New widgets to display properly

BrainBlasted commented:
Why was this closed?

Trogel commented:
@BrainBlasted Sorry, I should have left a comment here, as well. In the meantime I switched to LineageOS' built-in weather widget (see also my recent comment on #31). So, for the foreseeable future I will not contribute to Good Weather. To clean things up, I deleted my lingering branch, and so the PR got closed, too.

That should not stop anyone from using the suggested patch c2d8574 for a new PR. I still think that it would fix #31.
